x-pack/plugins/streams_app/public/util/use_debounce.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { debounce } from 'lodash';
+import { useCallback, useEffect, useState } from 'react';
+
+export function useDebounced<T>(value: T, debounceDelay: number = 300) {
```

Comment:
You might be able to use `useDebounce` from `react-use`.

Reply:
Good call, done!